### PR TITLE
Refactoring: Require return types in new ui

### PR DESCRIPTION
**Pull request by sven-urbanski-freiheit-com**
Added rule '@typescript-eslint/explicit-function-return-type' which requires all functions to have an explicit return type.
    
Also added a specific .eslintrc for the old ui, which disables this rule, because we do not want to spend much time with the old ui anymore.
    
Adapted all functions that didn't have a return type.
    
This is a refactoring, it does not change any behavior.

SRX-2DXE9T

### CODE REVIEW

File: services/frontend-service/src/ui/utils/store.tsx [hunk 1]
```diff
@@ -306,7 +307,7 @@ export const useEnvironmentLock = (lockId: string) =>
         )?.name,
     } as DisplayLock);
 
-export const searchCustomFilter = (queryContent: string | null, val: string | undefined) => {
+export const searchCustomFilter = (queryContent: string | null, val: string | undefined): string | undefined | null => {
```
Comment (tameremad) [inline services/frontend-service/src/ui/utils/store.tsx:310]: Can we handle the `undefined | null` cases here and return perhaps `''`?

File: services/frontend-service/src/ui/utils/store.tsx [hunk 2]
```diff
@@ -363,10 +364,10 @@ export const useApplicationLock = (lockId: string) =>
         )?.name,
     } as DisplayLock);
 
-export const useLock = (lockId: string) =>
+export const useLock = (lockId: string): DisplayLock | undefined =>
```
Comment (tameremad) [inline services/frontend-service/src/ui/utils/store.tsx:367]: Also the same comment as before? 

Additionally, to me, this function has no usages anywhere. do we need to keep it?

**Review by tameremad — APPROVED** [review]
Sounds good!

File: services/frontend-service/src/ui/utils/store.tsx (hunk 1 above)
Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/utils/store.tsx]: done

File: services/frontend-service/src/ui/utils/store.tsx (hunk 2 above)
Comment (sven-urbanski-freiheit-com) [reply services/frontend-service/src/ui/utils/store.tsx]: done, removed it.

**Pull request merged**